source/class/qx/io/jsonrpc/Client.js

```diff
@@ -125,28 +125,37 @@ qx.Class.define("qx.io.jsonrpc.Client", {
     },
 
     /**
-     * Fires "error" event and throws the error after informing pending requests
-     * about the error.
+     * Fires "error" event and rejects the pending requests' promises.
+     * The method will be renamed and made private in v8.  
      * @param exception
      * @private
      */
     _throwTransportException(exception) {
       this.fireDataEvent("error", exception);
       this.__requests.forEach(request => {
         if (request instanceof qx.io.jsonrpc.protocol.Request) {
+          // this rejects the request's promise
           request.handleTransportException(exception);
         }
       });
-      throw exception;
+      if (!qx.core.Environment.get("qx.io.jsonrpc.forwardTransportPromiseRejectionToRequest")){
+        throw exception; // will be removed in v8 since it is not caught anywhere
+      }
     },
 
     /**
      * Send the given JSON-RPC message object using the configured transport
      *
      * @param {qx.io.jsonrpc.protocol.Message|qx.io.jsonrpc.protocol.Batch} message
-     * @return {qx.Promise} Promise that resolves (with no data)
-     * when the message has been successfully sent out, and rejects
-     * when there is an error or a cancellation up to that point.
+     * @return {qx.Promise} Promise that resolves (with no data) when the message has been successfully 
+     * sent out. As this means different things depending on the transport implementation, it is best
+     * not to base any kind of business logic on the fulfillment of that promise.
+     * 
+     *  Rejections are forwarded 
+     * to the request's promise, which also fulfills with the jsonrpc response. Thus, there is no  need to 
+     * await the returned promise. Instead, await the request's promise, which and rejects either with a 
```

Comment:
"which and rejects"... Should that be "which fulfills and rejects" or "which rejects"?

Reply:
Oh darn I forgot to update the docs here, sorry. Will do when I fix the tests.

Reply:
Updated the docs. Also, we should now have a fully BC fix.